app/src/main/java/com/example/notes2text/adapters/FileEditingAdapters/FileEditorActivity.java

```diff
@@ -51,18 +56,29 @@ protected void onCreate(Bundle savedInstanceState) {
         original = textToEdit.getText();
 
         Bundle bundle = getIntent().getExtras();
-        // Receiving file from ActivitySwitchController
-        file = (File)bundle.getSerializable("file");
-        // Receiving array of selected files from JoinController
-        ArrayList<File> selectedFiles = (ArrayList<File>)bundle.getSerializable("selectedFiles");
-
-        // if received file from ActivitySwitchController
-        if (file != null) {
+        // Receiving file from ActivitySwitchController or SavePopUpActivity
```

Comment:
Thanks for adding in the explanations as per my last request - it is all clear now.